### ISSUE TITLE
Profilers do not work in threads

### ISSUE THREAD

**Issue by spacemanspiff2007** [python/cpython#124656]
# Bug report

### Bug description:

This issue is related to, but not the same as #110770.
Running two profilers in parallel threads raises an unexpected error.
It's not clear at all that the profiler will not work in parallel.
Either this is a bug or there should be a prominent warning in the docs.

```python
import time
from cProfile import Profile
from threading import Thread


def task_sleep() -> None:

    pr = Profile()
    pr.enable()
    time.sleep(1)
    pr.disable()


t1 = Thread(target=task_sleep)
t2 = Thread(target=task_sleep)

t1.start()
t2.start()

t1.join()
t2.join()
```

```
Exception in thread Thread-2 (task_sleep):
ValueError: Another profiling tool is already active
```

### CPython versions tested on:

3.12

### Operating systems tested on:

Windows

### Linked PRs
* gh-124671

**Comment by ZeroIntensity** [comment]
Yeah, you can't use multiple profilers at once. This is sort of documented, see [sys.monitoring.use_tool_id](https://docs.python.org/3/library/sys.monitoring.html#sys.monitoring.use_tool_id):

> Raises a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) if tool_id is in use.

I guess it's not explicitly mentioned in the `cProfile` documentation, though.
